src/speculators/models/dspark/metrics.py

```
@@ -81,10 +85,21 @@ def compute_metrics(
+    # [1, T, V] tensors (~3.2 GiB each at V=248320, T=3584) which OOM 60 GiB
+    # NPUs when four of them are live at once. Under no_grad nothing is saved
+    # for backward, so chunking is purely transient-buffer management; the ops
+    # are row-independent so the values are identical to the unchunked form.
     with torch.no_grad():
-        draft_p = softmax(logits.float(), dim=-1)
-        target_p = softmax(targets.float(), dim=-1)
-        accept_rate = torch.minimum(draft_p, target_p).sum(dim=-1)  # [1, T]
+        chunk = _ACCEPT_RATE_CHUNK
+        rates = []
+        for start in range(0, seq_len, chunk):
+            l = logits[:, start : start + chunk].float()
```

Ruff (0.16.2)

[error] Ambiguous variable name: `l`

(E741)